### PR TITLE
feat: CNS-validator-rewards-query

### PR DESCRIPTION
**Pull request by Yaroms**
# Description

Closes: #XXXX

---

## Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] read the [contribution guide](https://github.com/lavanet/lava/blob/main/CONTRIBUTING.md)
* [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title, you can find examples of the prefixes below:
    
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] targeted the `main` branch
* [ ] provided a link to the relevant issue or specification
* [ ] reviewed "Files changed" and left comments if necessary
* [ ] included the necessary unit and integration tests
* [ ] updated the relevant documentation or specification, including comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] confirmed all CI checks have passed

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic, API design and naming, documentation is accurate, tests and test coverage

## Summary by CodeRabbit

- **New Features**
	- Introduced new commands in the CLI for estimating rewards and tracking compute unit usage.
	- Added functionality for querying tracked information and rewards estimation in the subscription management context.
	- Enhanced API capabilities with new RPC methods for tracked usage and estimated rewards.

- **Bug Fixes**
	- Removed outdated methods which may cause confusion in reward queries.

- **Documentation**
	- Improved organizational structure in protocol buffer definitions for better clarity.

- **Chores**
	- Updated initialization parameters for keeper instances to enhance functionality.

### CODE REVIEW

File: x/subscription/keeper/grpc_query_estimated_validator_rewards.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	}
+
+	monthsLeft := k.rewardsKeeper.AllocationPoolMonthsLeft(ctx)
+	allocationPool := k.rewardsKeeper.TotalPoolTokens(ctx, rewardstypes.ValidatorsRewardsAllocationPoolName)
+	blockRewards := allocationPool.QuoInt(sdk.NewInt(monthsLeft))
+
+	iprpcReward, found := k.rewardsKeeper.GetIprpcReward(ctx, k.rewardsKeeper.GetIprpcRewardsCurrentId(ctx))
+	if found {
+		for _, fund := range iprpcReward.SpecFunds {
+			res.Info = append(res.Info, &types.EstimatedRewardInfo{Source: "iprpc_" + fund.Spec, Amount: sdk.NewDecCoinsFromCoins(fund.Fund...).MulDec(delegatorPart)})
```
Comment (omerlavanet) [inline x/subscription/keeper/grpc_query_estimated_validator_rewards.go:93]: missing tax factor

File: x/subscription/keeper/grpc_query_estimated_validator_rewards.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	}
+
+	valRewardsFromSubscriptions, _, err := k.rewardsKeeper.CalculateValidatorsAndCommunityParticipationRewards(ctx, totalSubsRewards[0])
+	if err != nil {
+		return nil, fmt.Errorf("failed to calculate Validators And Community Participation Rewards")
+	}
+
+	monthsLeft := k.rewardsKeeper.AllocationPoolMonthsLeft(ctx)
+	allocationPool := k.rewardsKeeper.TotalPoolTokens(ctx, rewardstypes.ValidatorsRewardsAllocationPoolName)
+	blockRewards := allocationPool.QuoInt(sdk.NewInt(monthsLeft))
```
Comment (omerlavanet) [inline x/subscription/keeper/grpc_query_estimated_validator_rewards.go:88]: reduce community pool part here

**Review by omerlavanet — CHANGES_REQUESTED** [review]

File: x/subscription/keeper/grpc_query_estimated_validator_rewards.go (hunk 2 above)
Comment (Yaroms) [reply x/subscription/keeper/grpc_query_estimated_validator_rewards.go:88]: done

**Review by omerlavanet — DISMISSED** [review]

File: x/rewards/keeper/iprpc_test.go [hunk 3]
```diff
@@ -472,7 +477,7 @@ func TestFundIprpcTwice(t *testing.T) {
 	// check rewards - should be only from first + second funding (=iprpcFunds*3)
 	res, err = ts.QueryDualstakingDelegatorRewards(p1Acc.GetVaultAddr(), p1, mockSpec2)
 	require.NoError(t, err)
-	require.True(t, iprpcFunds.Sub(minIprpcCost).MulInt(math.NewInt(3)).IsEqual(res.Rewards[0].Amount))
+	require.True(t, iprpcFunds.Sub(minIprpcCost).MulInt(math.NewInt(3)).MulInt(tax).QuoInt(sdk.NewInt(100)).IsEqual(res.Rewards[0].Amount))
 }
```
[the remark this answers is not in the document]
Comment (Yaroms) [reply x/rewards/keeper/iprpc_test.go:486]: done

**Review by omerlavanet — APPROVED** [review]

**Pull request merged**